tests/run/partial_circular_import.srctree

```diff
@@ -16,9 +16,25 @@ setup(
 ######## pkg/A.py ########
 from . import B
 
+def verify(rel_B):
+    import pkg.B
+    assert pkg.B == rel_B
+verify(B)
+
 ######## pkg/B.py ########
 from . import C
 
+def verify(rel_C):
+    import pkg.C
+    assert pkg.C == rel_C
+verify(C)
+
 ######## pkg/C.py ########
 from . import B
 
+def verify(rel_B):
+    import pkg.B
+    assert pkg.B == rel_B
+
+verify(B)
```

Comment:
I don't think this test can work (hence it fails on pure Python too) - the contents of `B` aren't filled in until after module `C` has run fully. I think maybe you should call verify up in the `PYTHON -c ...` lines after the import statement has completed fully.

Reply:
Interesting. I would have expected that to work…

Anyway, what I wanted to assure is that `__Pyx_Import()` returns the module and not the package.

I'm not even sure that we need `__Pyx_Import()` here. A lookup in `sys.modules` might be enough. Didn't take the time to dig up what exactly Python is doing here – in any case, we should mimic that.

Reply:
Having tested it in more detail: I think I misrepresented the problem with these tests slightly. It's actually that the contents of `pkg` aren't filled in until all its submodules are successfully reported

As you suspect, Python is just looking up in `sys.modules` - https://github.com/python/cpython/commit/0373a106a1ca9ce67cd99bf0a93ab7c3abd15cb6 (that's the implementation from 3.5 but the current version isn't notably different) so it's probably worth simplifying to that